src/MagnumPlugins/BasisImporter/Test/BasisImporterTest.cpp

```diff
@@ -0,0 +1,246 @@
+                       &BasisImporterTest::rgba},
+                       Containers::arraySize(FormatData));
+
+    /* Load the plugin directly from the build tree. Otherwise it's static and
+       already loaded. */
+    #ifdef BASISIMPORTER_PLUGIN_FILENAME
+    CORRADE_INTERNAL_ASSERT(_manager.load(BASISIMPORTER_PLUGIN_FILENAME) & PluginManager::LoadState::Loaded);
+    #endif
+}
+
+void BasisImporterTest::empty() {
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("BasisImporter");
+    importer->configuration().setValue("format", "Etc2");
+
+    std::ostringstream out;
+    Error redirectError{&out};
+    char a{};
+    /* Explicitly checking non-null but empty view */
+    CORRADE_VERIFY(!importer->openData({&a, 0}));
+    CORRADE_COMPARE(out.str(), "Trade::BasisImporter::openData(): the file is empty\n");
+}
+
+void BasisImporterTest::invalid() {
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("BasisImporter");
+    std::ostringstream out;
+    Error redirectError{&out};
+    CORRADE_VERIFY(!importer->openData("NotABasisFile"));
+
+    CORRADE_COMPARE(out.str(), "Trade::BasisImporter::openData(): invalid header\n");
+}
+
+void BasisImporterTest::unconfigured() {
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("BasisImporter");
+    std::ostringstream out;
+    Error redirectError{&out};
+
+    CORRADE_VERIFY(importer->openFile(
+        Utility::Directory::join(BASISIMPORTER_TEST_DIR, "rgb.basis")));
+    CORRADE_VERIFY(!importer->image2D(0));
+
+    CORRADE_COMPARE(out.str(), "Trade::BasisImporter::image2D(): invalid configuration: format is expected to be one of: Etc1, Etc2, Bc1, Bc3, Bc4, Bc5, Bc7M6OpaqueOnly, Pvrtc1_4OpaqueOnly\n");
+}
+
+void BasisImporterTest::openTwice() {
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("BasisImporter");
+    CORRADE_VERIFY(importer->openFile(
+        Utility::Directory::join(BASISIMPORTER_TEST_DIR, "rgb.basis")));
+    CORRADE_VERIFY(importer->openFile(
+        Utility::Directory::join(BASISIMPORTER_TEST_DIR, "rgb.basis")));
+
+    /* Shouldn't crash, leak or anything */
+}
+
+void BasisImporterTest::importTwice() {
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("BasisImporter");
+    CORRADE_VERIFY(importer->openFile(Utility::Directory::join(BASISIMPORTER_TEST_DIR, "rgb.basis")));
+
+    /* Verify that everything is working the same way on second use */
+    {
+        importer->configuration().setValue("format", "Etc2");
+
+        Containers::Optional<Trade::ImageData2D> image = importer->image2D(0);
+        CORRADE_VERIFY(image);
+        CORRADE_COMPARE(image->size(), (Vector2i{63, 27}));
+    } {
+        importer->configuration().setValue("format", "Bc1");
+
+        Containers::Optional<Trade::ImageData2D> image = importer->image2D(0);
+        CORRADE_VERIFY(image);
+        CORRADE_COMPARE(image->size(), (Vector2i{63, 27}));
+    }
+}
```

Comment:
I would expand this one to test also the imported format like the others do, call it `importMultipleFormats()` and move it below the rgb/rgba ones. Because how it is now it feels like a variant of the trivial `importTwice()` tests from elsewhere that's testing the target format switch only by accident.

Reply:
Ergo, rename and move `importTwice` and add format check, correct?

Reply:
Yes.